extra/modules/confiant-ad-quality/src/main/java/org/prebid/server/hooks/modules/com/confiant/adquality/v1/ConfiantAdQualityBidResponsesScanHook.java

```diff
@@ -94,6 +95,22 @@ private InvocationResult<AllProcessedBidResponsesPayload> toInvocationResult(
         final boolean hasIssues = bidsScanResult.hasIssues();
         final boolean debugEnabled = auctionInvocationContext.debugEnabled();
 
+        final List<BidderResponse> bidderResponsesWithIssues = new ArrayList<>();
+        List<BidderResponse> bidderResponsesWithoutIssues;
+
+        if (hasIssues) {
+            bidderResponsesWithoutIssues = new ArrayList<>();
+            for (int i = 0; i < scannedBidderResponses.size(); i++) {
+                if (bidsScanResult.hasIssuesByBidIndex(i)) {
+                    bidderResponsesWithIssues.add(scannedBidderResponses.get(i));
+                } else {
+                    bidderResponsesWithoutIssues.add(scannedBidderResponses.get(i));
+                }
+            }
```

Comment:
I'd say that the `bidsScanResult` should take responsibility of this piece of code, the only difference with the `filterValidResponses` method is this code returns both valid and invalid responses. In that way your can optimize the code even further because currently `hasIssues` iterates over all the responses also

Reply:
Good point. Thanks for a "side" view. Going to apply logic update.

Reply:
Wasn't sure about proper name. Hope current one is quite clear and updates can resolve your idea about optimisation.